datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/settings/applications/UpdateApplicationsSettingsResolver.java

```diff
@@ -0,0 +1,75 @@
+package com.linkedin.datahub.graphql.resolvers.settings.applications;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.*;
+
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.authorization.AuthorizationUtils;
+import com.linkedin.datahub.graphql.concurrency.GraphQLConcurrencyUtils;
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.generated.UpdateApplicationsSettingsInput;
+import com.linkedin.metadata.service.SettingsService;
+import com.linkedin.settings.global.GlobalSettingsInfo;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+/** Resolver responsible for updating the actions settings. */
+public class UpdateApplicationsSettingsResolver implements DataFetcher<CompletableFuture<Boolean>> {
+
+  private final SettingsService _settingsService;
+
+  public UpdateApplicationsSettingsResolver(@Nonnull final SettingsService settingsService) {
+    _settingsService = Objects.requireNonNull(settingsService, "settingsService must not be null");
+  }
+
+  @Override
+  public CompletableFuture<Boolean> get(final DataFetchingEnvironment environment)
+      throws Exception {
+    final QueryContext context = environment.getContext();
+    final UpdateApplicationsSettingsInput input =
+        bindArgument(environment.getArgument("input"), UpdateApplicationsSettingsInput.class);
+
+    return GraphQLConcurrencyUtils.supplyAsync(
+        () -> {
+          if (AuthorizationUtils.canManageFeatures(context)) {
+            try {
+              // First, fetch the existing global settings. This does a R-M-F.
+              final GlobalSettingsInfo maybeGlobalSettings =
+                  _settingsService.getGlobalSettings(context.getOperationContext());
+
+              final GlobalSettingsInfo newGlobalSettings =
+                  maybeGlobalSettings != null ? maybeGlobalSettings : new GlobalSettingsInfo();
+
+              final com.linkedin.settings.global.ApplicationsSettings newApplicationsSettings =
+                  newGlobalSettings.hasApplications()
+                      ? newGlobalSettings.getApplications()
+                      : new com.linkedin.settings.global.ApplicationsSettings().setEnabled(true);
```

Comment:
and we want this to be default enabled if there wasn't applications in global settings? i know it probably would get updated in `updateApplicationSettings` but the `enabled` field isn't required in graphql so it's possible that this gets set to true without the user changing `enabled`